### PR TITLE
feature/js_testing

### PR DESCRIPTION
**Pull request by kyfantaz**
-Moves js libraries into sub-directory `js/libraries` and refactors the templates to reference the new location for js assets
-Adds `karma.conf.js` file and `package.json` file
-Moves functions in `experiments_dashboard.js` outside of document.ready block to allow for testing
-Changes method `isValidUrl()` regex to require http(s) in url inputs
-Adds more js tests, renames js tests from naming convention `*_spec.js` to naming convention `test_*.js`, and moves location of js tests into `ab_tool/tests/frontend` from `spec/javascripts`

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    "karma": "^0.12.31",
+    "karma-jasmine": "^0.3.5",
+    "karma-phantomjs-launcher": "^0.1.4"
+  },
+  "devDependencies": {},
+  "scripts": {
+    "test": "echo \"Error: no test specified\" && exit 1"
+  },
+  "author": "kevinfan",
+  "license": "ISC",
```
Comment (elliottyates) [inline package.json]: @mmtchong Does this conflict in any way with the main LICENSE, which is MIT?

**Comment by elliottyates** [comment]
This looks good but I'd like some steps on how to test it -- e.g. what npm commands need to be run, etc. @mmtchong can we link to that documentation in the PR so I can proceed? Thanks!

File: package.json (hunk 1 above)
Comment (mmtchong) [inline package.json]: @kyfantaz is there a particular reason you're using an ISC license? Can you clarify?

**Comment by mmtchong** [comment]
@elliottyates Here are the links for the [overview] (https://drive.google.com/open?id=0BzkFrvsbF_2fWUQ5NGx6QTM3UTA) and [commands] (https://drive.google.com/open?id=0BzkFrvsbF_2fWUQ5NGx6QTM3UTA)

File: package.json (hunk 1 above)
Comment (kyfantaz) [inline package.json]: That line was auto-generated. You can change the license.

**Comment by elliottyates** [comment]
:+1: I'd be +2 if this PR included changes to the source code for (1) extra info in the README or documentation on how to run the tests, and (2) vagrant provisioner commands to ensure you don't have to run the manual installations in the VM (though I'm sure there are arguments for not doing that as well).

**Comment by elliottyates** [comment]
Ah, also @kyfantaz could you change the license to MIT please before merging?

**Pull request merged**